_posts/2021/2021-08-02-dont-call-replacement-classes-new.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Don't call a new version of something 'new' when refactoring"
+description: "A opinionated naming subtlety"
+layout: article
+category: ruby
+image:
+  base: '2021/dont-call-replacement-classes-new'
+  alt: "New"
+  credit: Nick Fewings
+  source: "https://unsplash.com/photos/1SsUquHPNT8"
+
+---
+
+In long-lived applications, we often want to rewrite or refactor an approach that we have previously taken.
+
+Here's a, non-obvious, naming technique I've used during refactoring to ensure that the improved code ends up in well named classes and any old implementation is clearly marked for deletion.
```

Comment:
*well-named

*..and that any old implementations are clearly marked for deletion.

Reply:
```suggestion
Here's a, non-obvious, naming technique I've used during refactoring to ensure that the improved code ends up in well-named classes and that any old implementation are clearly marked for deletion.
```